Cartesian_kernel/include/CGAL/constructions/kernel_ftC3.h

```diff
@@ -264,9 +264,20 @@ point_on_planeC3(const FT &pa, const FT &pb, const FT &pc, const FT &pd,
                  FT &x, FT &y, FT &z)
 {
   x = y = z = 0;
-  if (! CGAL_NTS is_zero(pa))      x = -pd/pa;
-  else if (! CGAL_NTS is_zero(pb)) y = -pd/pb;
-  else                             z = -pd/pc;
+  
+  FT abs_pa = CGAL::abs(pa);
+  FT abs_pb = CGAL::abs(pb);
+  FT abs_pc = CGAL::abs(pc);
+  
+  // to avoid badly defined point with an overly large coordinate when
+  //  the plane is almost orthogonal to one axis, we use the largest
+  //  scalar coordinate instead of always using the first non-null
+  if (CGAL::possibly(abs_pa >= abs_pb && abs_pa >= abs_pc))
```

Comment:
The use of `possibly` with `&&` seems strange to me. `&&` is not overloaded for `Uncertain` so it converts to bool, so `possibly` should be useless. I think you want either `possibly(a>=b)&&possibly(a>=c)` or `possibly(a>=b&a>=c)`.

Reply:
I wonder why `&&` and `||` are not overloaded for `Uncertain<bool>` (but only `&` and `|`). That would allow the writing of generic code deal either with `bool` or `Uncertain<bool>`. Actually, we have such generic code, and it uses the conversion to `bool`.

.. Well, the reason we do not have `&&`/`||` is actually documented:
https://github.com/CGAL/cgal/blob/e2f89318e57d9428e8a221a7d337bab4c5cd51d5/STL_Extension/include/CGAL/Uncertain.h#L338-L360

It was added ten years ago, in 449195eb9d15dac762e620104e2a076ce669203c.

Reply:
Ok, so I can use `CGAL_AND`. I suppose I also correct it in this other place: https://github.com/CGAL/cgal/pull/2122/commits/62df1454e3cefe26c03f9e5d2ca418d1eec95413